libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1262,18 +1250,25 @@ IotMqttError_t IotMqtt_Connect( const IotMqttNetworkInfo_t * pNetworkInfo,
         IotLogError( "Failed to establish new MQTT connection, error %s.",
                      IotMqtt_strerror( status ) );
 
-        /* The network connection must be closed if it was created. */
-        if( ownNetworkConnection == true )
+        if( status != IOT_MQTT_NETWORK_ERROR )
```

Comment:
Line 1130 also sets an `IOT_MQTT_NETWORK_ERROR` if the receive callback was not set. In this case, wouldn't the connection have been created but not destroyed by the cleanup code? If you kept the ownNetworkConnection bool, then changing line 1093's `EMPTY_ELSE_MARKER` to `ownNetworkConnection = pNetworkInfo->createNetworkConnection;` would allow most of this code block to stay as it was.

Reply:
I've reverted the change which eliminated `ownNetworkConnection`.